pkg/flowaggregator/flowaggregator.go

```diff
@@ -901,6 +902,13 @@ func (fa *flowAggregator) handleWatcherEvent() error {
 }
 
 func (fa *flowAggregator) updateFlowAggregator(opt *options.Options) {
+	// If user tries to change the mode dynamically, it makes sense to error out immediately and
+	// ignore other updates, as this is such a major configuration parameter.
+	// Unsupported "minor" updates are handled at the end of this function.
+	if opt.AggregatorMode != fa.aggregatorMode {
```

Comment:
I am wondering why not to introduce an annotation 'checksum/config' like Antrea Pods to make the FlowAggregator Pod auto-restart when the config is updated?

Reply:
That's a good point. The annotation should be here, but it's missing, so I'll open a separate PR to add it.
This code is for dynamic config changes (no Pod restart required). However, in retrospect, I am not so sure it was a good idea to add that support, given that not all config changes are supported. We could consider removing it.
Note that the annotation is not automatically updated when the ConfigMap is manually updated. But it does work well if the configuration is updated through Helm.